pkg/backup/backup.go

```diff
@@ -391,12 +415,24 @@ func (kb *kubernetesBackupper) BackupWithResolvers(log logrus.FieldLogger,
 	// no more progress updates will be sent on the 'update' channel
 	quit <- struct{}{}
 
-	// back up CRD for resource if found. We should only need to do this if we've backed up at least
-	// one item for the resource and IncludeClusterResources is nil. If IncludeClusterResources is false
-	// we don't want to back it up, and if it's true it will already be included.
-	if backupRequest.Spec.IncludeClusterResources == nil {
-		for gr := range backedUpGroupResources {
-			kb.backupCRD(log, gr, itemBackupper)
+	if collections.UseOldResourceFilters(backupRequest.Spec) {
+		// back up CRD for resource if found. We should only need to do this if we've backed up at least
+		// one item for the resource when IncludeClusterResources is nil. If IncludeClusterResources is false
+		// we don't want to back it up, and if it's true it will already be included.
+		if backupRequest.Spec.IncludeClusterResources == nil {
+			for gr := range backedUpGroupResources {
+				kb.backupCRD(log, gr, itemBackupper)
+			}
+		}
+	} else {
+		// back up CRD for resource if found. We should only need to do this if we've backed up at least
+		// one item for the resource when cluster-scoped resource filters are not specified.
+		// If cluster-scoped resource filters are set, whether or not to backup up the resource,
+		// depending on whether CRD is included by the filters setting.
+		if len(backupRequest.Spec.IncludedClusterScopeResources) == 0 && len(backupRequest.Spec.ExcludedClusterScopeResources) == 0 {
```

Comment:
Actually, we should include this as long as CRDs aren't listed explicitly in ExcludedClusterScopeResources (or includedClusterscopeResources). The length of the list doesn't matter here -- what matters is whether CRDs are listed. There are 3 ways that CRDs will be included here:
1) CRD is listed in ExcludedClusterScopeResources: don't include any CRDs, so skip this
2) CRD is listed in IncludedClusterScopeResources: we're including all CRDs in backup, so skip this case-by-case CRD inclusion
3) CRD listed in neither: this means include only relevant CRDs, so do it case-by-case.

Otherwise, a user explicitly including some other cluster-scoped resource would break including only relevant CRDs, which was one of the use cases we're trying to solve here.

Reply:
@sseago Thanks. I think the checking logic should be changed to CRD API resource couldn't be found in both ExcludedClusterScopeResources and IncludedClusterScopeResources.
It will cover both ClusterScopeResources filters using the default value and CRD is not listed in the filters.

Reply:
Modified.